### PR TITLE
Add JDK-20 support

### PR DESCRIPTION
**Pull request by reta**
### Description
Add JDK-20 support 
 
### Issues Resolved
Part of https://github.com/opensearch-project/ml-commons/issues/891
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@dhrubo-os should fix the `delombok` failures

**Review by b4sjoo — APPROVED** [review]

**Comment by b4sjoo** [comment]
Nice

**Comment by reta** [comment]
> Nice

Let me try to add JDK-20 to the matrix

**Review by b4sjoo — DISMISSED** [review]

**Comment by dhrubo-os** [comment]
> @dhrubo-os should fix the `delombok` failures

Thanks for raising the PR. Could you please raise the PR for 2.x branch first? Our most of the updated codes are in 2.x branch.

**Comment by b4sjoo** [comment]
2.x also moved to jdk20? @dhrubo-os

**Comment by dhrubo-os** [comment]
> 

Oh my bad. Forgot about that. @reta  NVM ignore my comment, lol

**Review by b4sjoo — APPROVED** [review]

File: common/build.gradle [hunk 1]
```diff
@@ -19,6 +19,10 @@ dependencies {
     testImplementation group: 'org.mockito', name: 'mockito-core', version: '4.4.0'
 }
 
+lombok {
```
Comment (dhrubo-os) [inline common/build.gradle:22]: Just curious: should we remove this: `id "io.freefair.lombok"`?

Comment (reta) [reply common/build.gradle:22]: > Just curious: should we remove this: `id "io.freefair.lombok"`?

The plugin is still needed, but it allows to override the lombok version (this is taken from the plugin docs)

**Review by dhrubo-os — APPROVED** [review]

**Pull request merged**